routes/originGitHub.js

```diff
@@ -20,13 +21,30 @@ router.get(
       const { login, repo } = request.params
       const github = getClient(request)
       const answer = await github.gitdata.getTags({ owner: login, repo, per_page: 100 })
-      const result = answer.data
-        .map(item => {
-          return { tag: item.ref.slice(10), sha: item.object.sha }
-        })
+      const unsorted = await Promise.all(answer.data
+        .map(throat(5, async item => {
+          if (item.object.type === 'commit')
+            return { tag: item.ref.slice(10), sha: item.object.sha }
+
+          if (item.object.type !== 'tag') {
+            return null
+          }
+
+          // we know now that we have an annotated tag, get the associated commit hash
+          let response
+          try {
+            response = await github.gitdata.getTag({ owner: login, repo, sha: item.object.sha })
+          } catch (e) {
+            console.error(e)
+            return null
+          }
+          return { tag: item.ref.slice(10), sha: response.data.object.sha }
```

Comment:
we should add a comment about this `slice(10)` something like

```
// slice /ref/tags/<tagname> to <tagname>
```

Reply:
Added a comment.